v1/messages/src/requestHandlers.ts

```diff
@@ -96,11 +107,47 @@ function parseSlugFromURI(uri: string, resource: string): string {
   return "";
 }
 
+function parseAuthorizationHeader(auth: string): string | null {
+  if (auth === undefined) {
+    return null;
+  }
+  if (auth.startsWith("Basic ")) {
+    let authID = auth.substring(6);
+
+    if (!validURLCharacters.test(authID)) {
+      return null;
+    }
+    return authID;
+  }
+  return null;
+}
+
+async function isUserAuth(
+  url: string,
+  authID: string | null
```

Comment:
Why not just parse the entire request object in this method, make `parseAuthorizationHeader` function private, and just call in here? That way, nobody outside this function needs to know anything about the authID, how it's parsed, etc.